### PR TITLE
Allow configuring x-forwarded-* proxy headers via nginx:set

### LINKED ISSUE #4339: Q: Resetting X-Forwarded-For header?
Hi,

I was wondering why the `X-Forwarded-For` header is being reset with the `$remote_addr` value in the [default Dokku nginx config](https://github.com/dokku/dokku/blob/e64b69248a4327bc41b3a3e1eb3736ccfb5b6bbb/plugins/nginx-vhosts/templates/nginx.conf.sigil#L100). This causes issues with fetching the client's IP address with an intermediate proxy server, like this example:

Client (1.1.1.1) --> Intermediate proxy (2.2.2.2) --> Dokku nginx --> Dokku app

In this case, the user's IP address in the Dokku app is reported as 2.2.2.2 (iso 1.1.1.1). This is because the Dokku nginx config overwrites any present `X-Forwarded-For` header with the current `$remote_addr` (2.2.2.2), rather than appending a second IP address.

An easy fix for this issue would be to replace $remote_addr with `$proxy_add_x_forwarded_for`.

P.s. Thank you for Dokku <3

### PR DESCRIPTION
**Pull request by bjornpost**
Fixes #4339

**Note:** If this PR is just doc changes, please put [ci skip] in the body that way tests do not run.

### CODE REVIEW

File: docs/configuration/ssl.md [hunk 1]
```diff
@@ -136,44 +136,9 @@ Certain versions of nginx have bugs that prevent [HTTP/2](https://nginx.org/en/d
...
-
-  location    / {
-    proxy_pass  http://{{ .APP }};
-    proxy_http_version 1.1;
-    proxy_set_header Upgrade $http_upgrade;
-    proxy_set_header Connection $http_connection;
-    proxy_set_header Host $http_host;
-    proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
-    proxy_set_header X-Forwarded-For $http_x_forwarded_for;
-    proxy_set_header X-Forwarded-Port $http_x_forwarded_port;
```
Comment (josegonzalez) [inline docs/configuration/ssl.md:157]: Should we do anything about all the other properties here? Would it make sense to instead make these values toggleable via an `nginx:set` call?

**Comment by josegonzalez** [comment]
Given your comment in the issue, seems like this shouldn't be default, and we should gate it behind an `nginx:set` flag.

File: docs/configuration/ssl.md (hunk 1 above)
Comment (bjornpost) [reply docs/configuration/ssl.md:157]: I don't think so. Only the value in `-For` is appended. `-Proto` and `-Port` are being replaced:

```bash
curl -H "X-Forwarded-Port: 1234" https://...

  ["HTTP_X_FORWARDED_PORT"]=>
  string(3) "443"
```

```bash
curl -H "X-Forwarded-Proto: Banana" https://...

  ["HTTP_X_FORWARDED_PROTO"]=>
  string(3) "https"
```

**Comment by bjornpost** [comment]
I've refactored the PR to only set `$http_x_forwarded_for` when configured via `nginx:set`. I don't have a test env locally, so I haven't tested the code yet. Thanks for the quick response!

File: plugins/nginx-vhosts/functions [hunk 2]
```diff
@@ -462,6 +462,8 @@ nginx_build_config() {
       local NGINX_BIND_ADDRESS_IP6="$(fn-plugin-property-get-default "nginx" "$APP" "bind-address-ipv6" "::")"
       [[ -z "$NGINX_BIND_ADDRESS_IP6" ]] && NGINX_BIND_ADDRESS_IP6="::"
 
+      local PROXY_TRUST_X_FORWARDED_FOR="$(fn-plugin-property-get-default "nginx" "$APP" "trust-x-forwarded-for" "false")"
```
Comment (josegonzalez) [inline plugins/nginx-vhosts/functions:465]: Can we add this to the nginx:report output?

File: tests/unit/test_helper.bash [hunk 3]
```diff
@@ -406,7 +406,7 @@ server {
     proxy_set_header Connection \$http_connection;
     proxy_set_header Host \$http_host;
     proxy_set_header X-Forwarded-Proto \$scheme;
-    proxy_set_header X-Forwarded-For \$remote_addr;
+    proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
```
Comment (josegonzalez) [inline tests/unit/test_helper.bash:409]: Should default back to `remote_addr`.

We also need a test of setting/unsetting this property to ensure it's set correctly/I dont break the option in a future refactor.

File: docs/configuration/ssl.md (hunk 1 above)
Comment (josegonzalez) [reply docs/configuration/ssl.md:157]: Yeah just thinking that removing this doc maybe isn't the best. Maybe we can add properties for each of these headers as well?

Comment (bjornpost) [reply docs/configuration/ssl.md:157]: I'm not sure if I understand you fully. I will undo the removal of the paragraph "Only use this option if ..." and the paragraph after. Do you also want to undo the deletion of the custom template example? I did add an example of how to configure via `nginx:set`, which makes more sense I think.

Regarding making `-Proto` and `-Port` toggleable: I can do it, but I'm not sure if its a common use case. Also not sure between which values you should be able to toggle.

Comment (josegonzalez) [reply docs/configuration/ssl.md:157]: If you're around on the gliderlabs slack, I'd be happy to jump on a quick call to explain what I mean. I'll be around for the next ~30 minutes.

**Comment by bjornpost** [comment]
Updated the PR again based on your feedback!

**Review by josegonzalez — APPROVED** [review]

**Comment by Haocen** [comment]
Sadly this is not working with `--global` flag.

**Pull request merged**